src/siteMap/mainSections.ts

```diff
@@ -1,48 +1,120 @@
 import { MainSectionType } from "../components/sideBar/menuItem/types";
-import { FaCloudUploadAlt, FaGamepad, FaPlay, FaTable } from "react-icons/fa";
+import {
+  FaCloudUploadAlt,
+  FaExternalLinkAlt,
+  FaGamepad,
+  FaPlay,
+  FaTable,
+} from "react-icons/fa";
 import { RiMovie2Line } from "react-icons/ri";
+import JsonPlaceHolderView from "../features/jsonPlaceholderAPI/JsonPlaceHolder";
+import PlaygroundView from "../features/playground/playgroundView";
+import ConfigTableExampleView from "../features/configTable/configTableExample";
+import MovieTableView from "../features/paginatedTable/moviePage";
+import VideoGamesTableView from "../features/formExample/videoGamesPage";
+import StaticPage from "../features/staticPage/staticPage";
 
-/*This is the Main Section ARRAY populate this to populate the main nav menu*/
+/*This is the Main Section ARRAY populate this to populate the main nav menu
+It is also used to create all the Routes for the router*/
 export const mainSections: MainSectionType[] = [
   {
-    path: "/main",
     label: "Main Test Section",
-    defaultSubMenuPath: "/manualUpload",
+
     subsections: [
       {
         path: "/jsonplaceholder",
         label: "Posts",
         icon: FaCloudUploadAlt,
         isInMenu: true,
         isExternal: false,
+        component: JsonPlaceHolderView,
       },
       {
         path: "/playground",
         label: "PlayGround",
         icon: FaPlay,
         isInMenu: true,
         isExternal: false,
+        component: PlaygroundView,
       },
       {
         path: "/configTable",
         label: "Config Table",
         icon: FaTable,
         isInMenu: true,
         isExternal: false,
+        component: ConfigTableExampleView,
       },
       {
         path: "/moviesTable",
         label: "Movie Paginated Table",
         icon: RiMovie2Line,
         isInMenu: true,
         isExternal: false,
+        component: MovieTableView,
       },
       {
         path: "/videoGamesTable",
         label: "VideoGames Table",
         icon: FaGamepad,
         isInMenu: true,
         isExternal: false,
+        component: VideoGamesTableView,
+      },
+    ],
+  },
+  {
+    label: "External Section",
+    subsections: [
+      {
+        externalUrl: "https://www.google.com",
+        label: "Google",
+        icon: FaExternalLinkAlt,
+        isInMenu: true,
+        isExternal: true,
+      },
+      {
+        externalUrl: "https://react.dev/",
+        label: "React",
+        icon: FaExternalLinkAlt,
+        isInMenu: true,
+        isExternal: true,
+      },
+    ],
+  },
+  {
+    label: "Another Section",
+
+    subsections: [
+      {
```

Comment:
subsection can have path. show example how to use them and route composition.

sections are useful to separate idential routes across... sections.

e.g.

/stuff 
    /list 
    /new

/otherstuff 
    /list 
    /new

Reply:
OK i used parent path to get all the page and component unique paths across sections